appcelerator-archive/amp
# Fixing vendors

Fix https://techweb.axway.com/jira/browse/AMP-98

This PR fixes vendor (again).

The version for each dependency has been fixed so that we're less vulnerable to any transitive dependency update.
The code has been updated to comply with the new dependencies.

## How to test
Make sure you have the latest release version of dep (0.3.2):
```
$ dep version
dep:
 version     : v0.3.2
 build date  : 2017-10-19
 git hash    : 8ddfc8a
 go version  : go1.9
 go compiler : gc
 platform    : linux/amd64
```

Test with the following instructions:
```
$ rm -rf Gopkg.lock vendor
$ dep ensure
$ dep prune
```

Note that your vendor folder may differ from the one committed on this branch. It's probably because transitive dependencies might have been updated in the meantime. For more details on how `dep` works, please read https://github.com/golang/dep/blob/master/docs/FAQ.md#how-do-i-constrain-a-transitive-dependencys-version

Then make sure the code compiles and tests as expected:
```
$ ampmake clean build
$ amp cluster create
$ export PATH=tests/cli/testing:$PATH
$ TESTINCLUDE=tests/cli/testing testrunner tests/cli
$ docker run -it --rm --network core -v $PWD:/go/src/github.com/appcelerator/amp -w /go/src/github.com/appcelerator/amp appcelerator/amptools go test github.com/appcelerator/amp/tests/integration/... 
```

Check Travis build.
